frontend/components/SingleDocument.js

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useState} from "react";
+import * as PropTypes from "prop-types";
+import STYLES from "./SingleDocument.module.scss";
+
+const SingleDocument = ({id}) => {
+
+    const [docData, setDocData] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [showText, setShowText] = useState(false);
+
+    useEffect(() => {
+        fetch(`/api/document/${id}`)
+            .then(response => response.json())
+            .then(data => {
+                setDocData(data);
+                setLoading(false);
+                console.log(data);
+            });
+    }, []);
+
+    const handleShowText = () => setShowText(!showText);
```

Comment:
The React Hook `setState` functions follow the same pattern as the class-based `setState` in that, if you need to set a state based on its previous value, it's best to use the updater function pattern, i.e.:
```
const handleShowText = () => setShowText(prevShowText => !prevShowText);
```
This ensures that the asynchrynous `setState` doesn't accidentally access the wrong value when it finally exists its queue.